pkg/customsignaturestore/customsignaturestore.go

```diff
@@ -58,44 +65,56 @@ func (s *Store) Signatures(ctx context.Context, name string, digest string, fn s
 		return done, err
 	}
 
-	stores := make([]store.Store, 0, len(uris))
-	for i := range uris {
-		uri := *uris[i]
-		stores = append(stores, &sigstore.Store{
-			URI:        &uri,
-			HTTPClient: s.HTTPClient,
-		})
+	stores := make([]store.Store, 0, len(customStores))
+	for _, customStore := range customStores {
+		uri, err := url.Parse(customStore.URL)
+		if err != nil {
+			return err
+		}
+		newHttpClient, err := s.HTTPClient(customStore.CA.Name)
+		if err != nil {
+			return err
+		}
+		if customStore.CA.Name != "" {
+			stores = append(stores, &sigstore.Store{
+				URI:        uri,
+				HTTPClient: func() (*http.Client, error) { return newHttpClient, nil }})
+		}
+
+	}
+	if len(stores) == 0 {
+		return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature")
 	}
 	store := &parallel.Store{Stores: stores}
 	if err := store.Signatures(ctx, name, digest, wrapper); err != nil || allDone {
 		return err
 	}
-	return errors.New("ClusterVersion spec.signatureStores exhausted without finding a valid signature.")
+	return nil
```

Comment:
This change does not seem correct, getting here means we haven't found a signature after exhausting all locations, I believe. The case when we find something is handled by the `allDone` condition on L89.

It is a bit troubling that this does not make any tests fail. Perhaps we are missing coverage.

Reply:
fixed in 4ead92af5e; I'm marking this thread resolved.  But feel free to spin out the test-coverage point to a new thread if you like.